src/features/expenses/AddExpense/AddParticipants.tsx

```diff
@@ -0,0 +1,161 @@
+import Header from "components/Header";
+import { newTransactionSelector, saveTransaction } from "../slice";
+import { ButtonBase, HalfCircle, TextUnderline } from "components";
+import defaultTheme from "constants/theme/default.theme";
+import { useGetTransactionGroupByIdQuery } from "features/groups/api";
+import SplitMethodButton from "./SplitMethods/SplitMethodButton";
+import EquallySplitMethodView from "./SplitMethods/Equally";
+import UnequallySplitMethodView from "./SplitMethods/Unequally";
+import AdjustmentSplitMethodView from "./SplitMethods/Adjustment";
+import PercentageSplitMethodView from "./SplitMethods/Percentage";
+import { ICreateTransaction, SplitMethod } from "../types";
+import { ITransactionGroup } from "features/groups/types";
+import { useCreateTransactionMutation } from "../api";
+
+import { AddParticipantsWrapper, BackgroundFiller, MovingBorder, SaveButtonWrapper, BorderShift, SplitMethodBorderWrapper, SplitMethodWrapper } from "./AddParticipants.styled";
+import Loader from "components/Loader";
+import toast from "react-hot-toast";
+import { composeGroupRoute } from "constants/routes";
+
+const GetSplitModeView = (splitMethodIndex: number, transaction: ICreateTransaction, group: ITransactionGroup) => {
+  if (splitMethodIndex === 0)
+    return <EquallySplitMethodView transaction={transaction} group={group} />
+  else if (splitMethodIndex === 1)
+    return <UnequallySplitMethodView transaction={transaction} group={group} />
+  else if (splitMethodIndex === 2)
+    return <PercentageSplitMethodView transaction={transaction} group={group} />
+  else if (splitMethodIndex === 3)
+    return <AdjustmentSplitMethodView transaction={transaction} group={group} />
+  else
+    throw new Error('Invalid split method index');
+}
+
+const splitMethodIndexToEnum = (splitMethodIndex: number) => {
+  if (splitMethodIndex === 0)
+    return SplitMethod.Equal;
+  else if (splitMethodIndex === 1)
+    return SplitMethod.AssignedAmount;
+  else if (splitMethodIndex === 2)
+    return SplitMethod.Percentage;
+  else if (splitMethodIndex === 3) // Ajustment uses Equal as base type and AssignedAmount for adjustments
+    return SplitMethod.Equal;
+  else
+    throw new Error('Invalid split method index');
+};
+   
+
+const AddParticipants = () => {
+  const { t } = useTranslation('expenses');
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+
+  const [createTransaction, { data: createdTransaction, isLoading: isTransactionCreating }] = useCreateTransactionMutation();
+
+  const transaction = useSelector(newTransactionSelector);
+  const group = useGetTransactionGroupByIdQuery(transaction?.groupId || '');
+
+  const [splitMethodIndex, setSplitMethodIndex] = useState<number>(0);
+
+  const handleBackAction = useCallback(() => {
+    navigate(-1);
+  }, [navigate]);
+
+  const resetPartialsAssignments = useCallback((splitIndex: number) => {
+    const newSplitMethod = splitMethodIndexToEnum(splitIndex);
```

Comment:
Try to implement something like SplitMethod[splitIndex]

Reply:
It was changed in 81508842d73fd52a080d5628b9689b50a2437d2d, please have a look.